src/main/scala/org/broadinstitute/dsde/agora/server/webservice/handlers/QueryHandler.scala

```diff
@@ -58,10 +60,22 @@ class QueryHandler(dataSource: PermissionsDataSource, implicit val ec: Execution
             entity: AgoraEntity,
             entityTypes: Seq[AgoraEntityType.EntityType],
             username: String,
-            onlyPayload: Boolean): Future[PerRequestMessage] = {
+            onlyPayload: Boolean,
+            payloadAsObject: Boolean): Future[PerRequestMessage] = {
     agoraBusiness.findSingle(entity, entityTypes, username: String) map { foundEntity =>
-      if (onlyPayload) RequestComplete(foundEntity.payload)
-      else RequestComplete(foundEntity)
+      if (onlyPayload && payloadAsObject) {
+        RequestComplete(BadRequest, "onlyPayload, payloadAsObject cannot be used together")
+      } else {
+
+        if (onlyPayload) {
+          RequestComplete(foundEntity.payload)
+        } else if (payloadAsObject) {
+          RequestComplete(foundEntity.deserializeConfigurationPayload)
+        } else {
+          RequestComplete(foundEntity)
+        }
```

Comment:
These nested ifs seem like a good case for a sequence of case statements

Reply:
How does
```
(onlyPayload, payloadAsObject) match {
  case (true, true) => RequestComplete(BadRequest, "onlyPayload, payloadAsObject cannot be used together")
  case (true, false) => RequestComplete(foundEntity.payload)
  case (false, true) => RequestComplete(foundEntity.deserializeConfigurationPayload)
  case _ => RequestComplete(foundEntity)
}
```
strike you? Lets the case statement handle the boolean logic for us.